Author: Iustin Pop <iusty@k1024.org>
Description: Fix the inet_addr result check with correct type casting
--- a/cbtcommon/tcpsocket.c
+++ b/cbtcommon/tcpsocket.c
@@ -198,7 +198,7 @@
     memcpy(dest, &ip.s_addr, sizeof(ip.s_addr));
   }
 #else
-  if ( (*dest = inet_addr(addr_str)) != -1)
+  if ( (*dest = inet_addr(addr_str)) != (in_addr_t)-1)
   {
     /* nothing */
   }
